config/rbac/net_attach_def_role.yaml

```diff
@@ -0,0 +1,22 @@
+---
+kind: ClusterRole
+metadata:
+  name: net-attach-def-editor
+  labels:
+    app.kubernetes.io/name: role
+    app.kubernetes.io/instance: net-attach-def-editor-role
+    app.kubernetes.io/component: rbac
+    app.kubernetes.io/created-by: selfserviceoverlay
+    app.kubernetes.io/part-of: selfserviceoverlay
+    app.kubernetes.io/managed-by: kustomize
+rules:
+- apiGroups:
+  - k8s.cni.cncf.io
+  resources:
+  - network-attachment-definitions
+  verbs:
+  - get
+  - list
+  - watch
```

Comment:
The controller is planned to create and probably delete NAD objects. Why does it need the other verbs?

Reply:
I think having create/delete permissions w/o get doesn't scale, we may need to verify the object is not already exist or has a unique spec before creation.
Or verifying it been disposed on deletion.

Reply:
I think it is unsafe for this controller to be able to list, watch or get arbitrary NADs.
Creating and deleting in a must, so we have no choice.

As an async & declarative system, I do not think we need to care if something already exists or is gone, we just try until it succeeds (e.g. for deletion, we retry until we get an error of NoExist or something similar. And on creation, we override if already exists.

But if I am wrong, you can re-introduce it. Adding the permissions before showing they are really needed is problematic.

Reply:
OK, I will remove them for now